_pytest/main.py

```diff
@@ -729,6 +729,25 @@ def _tryconvertpyarg(self, x):
 
         """
         import pkgutil
+
+        if six.PY2:  # python 3.4+ uses importlib instead
+            def find_module_patched(self, fullname, path=None):
+                subname = fullname.split(".")[-1]
+                if subname != fullname and self.path is None:
+                    return None
+                if self.path is None:
+                    path = None
+                else:
+                    path = [self.path]
+                try:
+                    file, filename, etc = pkgutil.imp.find_module(subname,
+                                                                  path)
+                except ImportError:
+                    return None
+                return pkgutil.ImpLoader(fullname, file, filename, etc)
+
+            pkgutil.ImpImporter.find_module = find_module_patched
```

Comment:
I'm a little concerned about patching this globally like it is done here... who knows what we might break by patching this.

I would rather be conservative here and patch this only while we are calling `find_loader` in the lines below this one. I think a nice way to implement this would be by using a `contextmanager`:

```python
try:
    with _patched_find_module():
        loader = pkgutil.find_loader(x)
except ImportError:
```

And `_patched_find_module` can be implemented like so:


```python


@contextlib.contextmanager
def _patched_find_module():
    """ ... """ 
    if six.PY2:  # python 3.4+ uses importlib instead
        def find_module_patched(self, fullname, path=None):
            subname = fullname.split(".")[-1]
            if subname != fullname and self.path is None:
                return None
            if self.path is None:
                path = None
            else:
                path = [self.path]
            try:
                file, filename, etc = pkgutil.imp.find_module(subname,
                                                              path)
            except ImportError:
                return None
            return pkgutil.ImpLoader(fullname, file, filename, etc)

        old_find_module = pkgutil.ImpImporter.find_module
        pkgutil.ImpImporter.find_module = find_module_patched
        try:            
            yield
        finally:
            pkgutil.ImpImporter.find_module = old_find_module
    else:
        yield

```

We can then even use `_patched_find_module()`'s docstring to better explain what's going on.